language/testing-infra/transactional-test-runner/src/framework.rs

```diff
@@ -675,24 +676,32 @@ where
     )
     .unwrap();
 
-    let first_task = tasks.pop_front().unwrap();
-    let init_opt = match &first_task.command {
-        TaskCommand::Init(_, _) => Some(first_task.map(|known| match known {
-            TaskCommand::Init(command, extra_args) => (command, extra_args),
-            _ => unreachable!(),
-        })),
-        _ => {
-            tasks.push_front(first_task);
-            None
-        }
+    let mut ctx = jpst::TemplateContext::new();
+    let first_lazy_task = tasks.pop_front().unwrap();
+    let first_task = first_lazy_task.parse(&ctx)?;
+    let (first_task, init_opt) = match &first_task.command {
+        TaskCommand::Init(_, _) => (
+            None,
+            Some(first_task.map(|known| match known {
+                TaskCommand::Init(command, extra_args) => (command, extra_args),
+                _ => unreachable!(),
+            })),
+        ),
+        _ => (Some(first_task), None),
     };
     let (mut adapter, result_opt) =
         Adapter::init(default_syntax, fully_compiled_program_opt, init_opt);
     if let Some(result) = result_opt {
         writeln!(output, "\ninit:\n{}", result)?;
     }
+
+    if let Some(first_task) = first_task {
+        handle_known_task(&mut output, &mut adapter, &mut ctx, first_task);
```

Comment:
Why is the first task different here?

Reply:
The first task may be the `Init` command.
In the previous version, we pop up the first task and check if it is an `Init` command. If not, we `push_front`.

In the new version, the first task is a LazyTask, we parse it, and it becomes a Task, and we can not `push_front`.